### PR TITLE
Fix SprintfEngine to allow more complex formatting as shown in readme example

### PR DESCRIPTION
**Pull request by tedswag**
The example in the readme shows the SprintfEngine being used to format to a number of decimal places, however, the regex used in the SprintfEngine was only matching one character. This meant any additional formatting beyond the type specifier was not being parsed.

A related unit test has been added, based off the readme example, and the regex changed so it now passes.

Additionally, the readme example did not correctly conform to the sprintf format so this has also been changed.

### CODE REVIEW

**Comment by tedswag** [comment]
I have also added PHPUnit to the composer.json file to allow contributors to easily run unit tests after running `composer install`.

File: tests/StringTemplate/SprintfEngineTest.php [hunk 1]
```diff
@@ -44,12 +44,27 @@ public function testRenderWithObjectValues()
...
     }
+
+    public function testRenderWithAdditionalFormatting()
+    {
+        $engine = new SprintfEngine();
+        $this->assertEquals(
+            'I have 1.2 (1.230000E+0) apples.',
+            $engine->render(
+                "I have {num%.1f} ({num%.6E}) {fruit}.",
+                [
```
Comment (nicmart) [inline tests/StringTemplate/SprintfEngineTest.php]: We need to mantain php 5.3 compatibility even at the test level, can you please fix that? Thanks!

**Comment by nicmart** [comment]
Thank you! Just waiting for a fix for tests in php 5.3 (see foregoing comments) and then the pull request can be merged into master and tagged.

**Comment by tedswag** [comment]
Should now be compatible.

**Comment by nicmart** [comment]
Great, can you please fix the conflict on the composer.json?
Thanks!

**Comment by nicmart** [comment]
Merged, tagged with version 0.1.1

**Pull request merged**